### ISSUE TITLE
Make sure Byron block generation is feature complete and tested

### ISSUE THREAD

**Issue by mrBliss** [input-output-hk/ouroboros-network#1194]
In `Ouroboros.Consensus.Ledger.Byron.Forge.forgeBlock`, we forge a block that can contain transactions, delegation certificates, update votes and proposals. Since we don't have a Byron transaction generator #888, this code hasn't been properly tested yet, as we only test with empty blocks. This should be done before we let the new node forge new blocks.

**Comment by mrBliss** [comment]
See more details in https://github.com/input-output-hk/ouroboros-network/issues/888#issuecomment-541607125

**Comment by edsko** [comment]
WIP PR to run Byron with dual ledgers at https://github.com/input-output-hk/ouroboros-network/pull/1425 .

**Comment by edsko** [comment]
Testing with delegation and update proposals will depend on https://github.com/input-output-hk/ouroboros-network/issues/1495 .

**Comment by edsko** [comment]
WIP PR to generate Byron transactions at https://github.com/input-output-hk/ouroboros-network/pull/1503 .

**Comment by edsko** [comment]
#1503 extends the DualPBFT tests with regular transactions. Opened a separate ticket about adding tests for update proposals in #1514, and a lower priority ticket for merging these two tests in #1515.
